bug: error while parsing aclfile

**Describe the bug**

Receiving an error `acl_family.cc:271 Error while parsing aclfile: Unrecognized parameter >mypassword` while using an known valid ACL file. 

```bash
"USER local-testing.rot-1 ON >mypassword resetkeys ~mykey* +set +get +del +ping"
``` 

**To Reproduce**
1. Create an ACL file with the contents above
2. Deploy on Kubernetes with Dragonfly DB operator v.1.1.2
3. View the logs of the StatefulSet
4. See error: `acl_family.cc:271 Error while parsing aclfile`

**Expected behavior**
File should be loaded successfully, according to your [documentation](https://www.dragonflydb.io/docs/managing-dragonfly/acl#passwords) `>` should be supported.

**Environment (please complete the following information):**
 - Containerized?: Kubernetes
 - Dragonfly Version: v1.17.1

**Additional context**
The bug may have been introduced here: https://github.com/dragonflydb/dragonfly/pull/2123

According to the [Redis Documentation](https://redis.io/docs/latest/operate/oss_and_stack/management/security/acl/#acl-rules
) `>` is used for plaintext passwords where as `#` is used for a hash value.  
```
Configure valid passwords for the user:

><password>: Add this password to the list of valid passwords for the user. For example >mypass will add "mypass" to the list of valid passwords. This directive clears the nopass flag (see later). Every user can have any number of passwords.
<<password>: Remove this password from the list of valid passwords. Emits an error in case the password you are trying to remove is actually not set.
#<hash>: Add this SHA-256 hash value to the list of valid passwords for the user. This hash value will be compared to the hash of a password entered for an ACL user. This allows users to store hashes in the acl.conf file rather than storing cleartext passwords. Only SHA-256 hash values are accepted as the password hash must be 64 characters and only contain lowercase hexadecimal characters.
```

**Non Relevant**
This is a really cool project and I appreciate all the hard work that has gone into it.  Keep up the great work!

--- a/src/server/acl/helpers.cc
+++ b/src/server/acl/helpers.cc
@@ -116,12 +116,11 @@ std::optional<std::string> MaybeParsePassword(std::string_view command, bool has
     return std::string(command);
   }
 
-  char symbol = hashed ? '#' : '>';
-  if (command[0] != symbol) {
-    return {};
+  if (command[0] == '>' || (hashed && command[0] == '#')) {
+    return std::string(command.substr(1));
   }
 
-  return std::string(command.substr(1));
+  return {};
 }
 
 std::optional<bool> MaybeParseStatus(std::string_view command) {
@@ -231,7 +230,9 @@ std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(T args,
         return ErrorReply("Only one password is allowed");
       }
       req.password = std::move(pass);
-      req.is_hashed = hashed;
+      if (hashed && absl::StartsWith(facade::ToSV(arg), "#")) {
+        req.is_hashed = hashed;
+      }
       continue;
     }
 

--- a/src/server/acl/helpers.h
+++ b/src/server/acl/helpers.h
@@ -23,6 +23,7 @@ std::string AclCommandToString(const std::vector<uint64_t>& acl_category);
 
 std::string PrettyPrintSha(std::string_view pass, bool all = false);
 
+// When hashed is true, we allow passwords that start with both # and >
 std::optional<std::string> MaybeParsePassword(std::string_view command, bool hashed = false);
 
 std::optional<bool> MaybeParseStatus(std::string_view command);

--- a/tests/dragonfly/acl_family_test.py
+++ b/tests/dragonfly/acl_family_test.py
@@ -323,11 +323,18 @@ async def test_bad_acl_file(df_local_factory, tmp_dir):
 @pytest.mark.asyncio
 @dfly_args({"port": 1111})
 async def test_good_acl_file(df_local_factory, tmp_dir):
-    acl = create_temp_file("", tmp_dir)
+    acl = create_temp_file("USER MrFoo ON >mypass", tmp_dir)
     df = df_local_factory.create(aclfile=acl)
 
     df.start()
     client = df.client()
+
+    await client.execute_command("ACL LOAD")
+    result = await client.execute_command("ACL LIST")
+    assert 2 == len(result)
+    assert "user MrFoo on ea71c25a7a60224 +@NONE" in result
+    assert "user default on nopass +@ALL +ALL ~*" in result
+    await client.execute_command("ACL DELUSER MrFoo")
 
     await client.execute_command("ACL SETUSER roy ON >mypass +@STRING +HSET")
     await client.execute_command("ACL SETUSER shahar >mypass +@SET")

